lib/deliver/upload_screenshots.rb

```diff
@@ -22,27 +22,34 @@ def upload(options, screenshots)
 
       # Now, fill in the new ones
       indized = {} # per language and device type
-      screenshots.each do |screenshot|
-        indized[screenshot.language] ||= {}
-        indized[screenshot.language][screenshot.device_type] ||= 0
-        indized[screenshot.language][screenshot.device_type] += 1 # we actually start with 1... wtf iTC
 
-        index = indized[screenshot.language][screenshot.device_type]
+      screenshots_per_language = screenshots.group_by {|s| s.language }
```

Comment:
Space between { and | missing.<br>Pass `&:language` as an argument to `group_by` instead of a block.